aptos-move/block-executor/src/captured_reads.rs

```diff
@@ -779,15 +872,18 @@ where
         use MVDataError::*;
         use MVDataOutput::*;
         self.data_reads.iter().all(|(k, r)| {
+            // We use fetch_data even with BlockSTMv2, because we don't want to record reads.
```

Comment:
Hmmm, so we just validate but make sure we do not add reads to write's read-set? Great! Maybe the top-level documentation comment for the method itself can mention that this does not record anything for Block-STM

Reply:
Yep, this is for V1 validation or the post-commit final safety checking validation of everything I added to materialization which runs for both V1 and V2 (and didn't cause perf regression as far as we could tell :). And using V1 interface makes sure we don't make the records. Will add an explicit comment

Reply:
That's a great suggestion. Descriptive from the API point of view, and the V1 / V2 choose what to call based on functionality. Thanks! I'm doing it for the reads for now, write and remove will be _v2, unless we think of some good naming suggestions there as well (but there also it's never mixed like here when V2 was calling v1 interface to avoid recording).